> static tests are failing :(
@eladkal

I guess this was a just random failure.. 

[error log]
```
An unexpected error has occurred: CalledProcessError: command: ('/opt/pipx/venvs/apache-airflow-breeze/bin/python', '-mnodeenv', '--prebuilt', '--clean-src', '/home/runner/.cache/pre-commit/repoui20kj4a/node_env-18.6.0', '-n', '18.6.0')
return code: 1
stdout: (none)
stderr:
     * Install prebuilt node (18.6.0) .Incomplete read while readingfrom https://nodejs.org/download/release/v18.6.0/node-v18.6.0-linux-x64.tar.gz
    .
    Traceback (most recent call last):
      File "/opt/hostedtoolcache/Python/3.7.15/x64/lib/python3.7/runpy.py", line 193, in _run_module_as_main
        "__main__", mod_spec)
      File "/opt/hostedtoolcache/Python/3.7.15/x64/lib/python3.7/runpy.py", line 85, in _run_code
        exec(code, run_globals)
      File "/opt/pipx/venvs/apache-airflow-breeze/lib/python3.7/site-packages/nodeenv.py", line 1519, in <module>
        main()
      File "/opt/pipx/venvs/apache-airflow-breeze/lib/python3.7/site-packages/nodeenv.py", line 1104, in main
        create_environment(env_dir, args)
      File "/opt/pipx/venvs/apache-airflow-breeze/lib/python3.7/site-packages/nodeenv.py", line 980, in create_environment
        install_node(env_dir, src_dir, args)
      File "/opt/pipx/venvs/apache-airflow-breeze/lib/python3.7/site-packages/nodeenv.py", line 739, in install_node
        install_node_wrapped(env_dir, src_dir, args)
      File "/opt/pipx/venvs/apache-airflow-breeze/lib/python3.7/site-packages/nodeenv.py", line [76](https://github.com/apache/airflow/actions/runs/4283958493/jobs/7460518056#step:6:77)2, in install_node_wrapped
        download_node_src(node_url, src_dir, args)
      File "/opt/pipx/venvs/apache-airflow-breeze/lib/python3.7/site-packages/nodeenv.py", line 602, in download_node_src
        with ctx as archive:
      File "/opt/hostedtoolcache/Python/3.7.15/x64/lib/python3.7/contextlib.py", line 112, in __enter__
        return next(self.gen)
      File "/opt/pipx/venvs/apache-airflow-breeze/lib/python3.7/site-packages/nodeenv.py", line 573, in tarfile_open
        tf = tarfile.open(*args, **kwargs)
      File "/opt/hostedtoolcache/Python/3.7.15/x64/lib/python3.7/tarfile.py", line 1573, in open
        saved_pos = fileobj.tell()
    AttributeError: 'bytes' object has no attribute 'tell'
    
Check the log at /home/runner/.cache/pre-commit/pre-commit.log
There were errors during pre-commit check. They should be fixed
Error: Process completed with exit code 3.
```

--- a/airflow/providers/google/cloud/hooks/bigquery_dts.py
+++ b/airflow/providers/google/cloud/hooks/bigquery_dts.py
@@ -304,11 +304,16 @@ class AsyncBiqQueryDataTransferServiceHook(GoogleBaseAsyncHook):
         sync_hook = await self.get_sync_hook()
         return sync_hook.project_id
 
+    async def _get_project_location(self) -> str:
+        sync_hook = await self.get_sync_hook()
+        return sync_hook.location
+
     async def get_transfer_run(
         self,
         config_id: str,
         run_id: str,
         project_id: str | None,
+        location: str | None = None,
         retry: Retry | _MethodDefault = DEFAULT,
         timeout: float | None = None,
         metadata: Sequence[tuple[str, str]] = (),
@@ -321,6 +326,7 @@ class AsyncBiqQueryDataTransferServiceHook(GoogleBaseAsyncHook):
         :param project_id: The BigQuery project id where the transfer configuration should be
             created. If set to None or missing, the default project_id from the Google Cloud connection
             is used.
+        :param location: BigQuery Transfer Service location for regional transfers.
         :param retry: A retry object used to retry requests. If `None` is
             specified, requests will not be retried.
         :param timeout: The amount of time, in seconds, to wait for the request to
@@ -330,8 +336,13 @@ class AsyncBiqQueryDataTransferServiceHook(GoogleBaseAsyncHook):
         :return: An ``google.cloud.bigquery_datatransfer_v1.types.TransferRun`` instance.
         """
         project_id = project_id or (await self._get_project_id())
+        location = location or (await self._get_project_location())
+        name = f"projects/{project_id}"
+        if location:
+            name += f"/locations/{location}"
+        name += f"/transferConfigs/{config_id}/runs/{run_id}"
+
         client = await self._get_conn()
-        name = f"projects/{project_id}/transferConfigs/{config_id}/runs/{run_id}"
         transfer_run = await client.get_transfer_run(
             name=name,
             retry=retry,

--- a/airflow/providers/google/cloud/operators/bigquery_dts.py
+++ b/airflow/providers/google/cloud/operators/bigquery_dts.py
@@ -306,6 +306,10 @@ class BigQueryDataTransferServiceStartTransferRunsOperator(GoogleCloudBaseOperat
 
     def execute(self, context: Context):
         self.log.info("Submitting manual transfer for %s", self.transfer_config_id)
+
+        if self.requested_run_time and isinstance(self.requested_run_time.get("seconds"), str):
+            self.requested_run_time["seconds"] = int(self.requested_run_time["seconds"])
+
         response = self.hook.start_manual_transfer_runs(
             transfer_config_id=self.transfer_config_id,
             requested_time_range=self.requested_time_range,

--- a/airflow/providers/google/cloud/triggers/bigquery_dts.py
+++ b/airflow/providers/google/cloud/triggers/bigquery_dts.py
@@ -101,6 +101,7 @@ class BigQueryDataTransferRunTrigger(BaseTrigger):
                     project_id=self.project_id,
                     config_id=self.config_id,
                     run_id=self.run_id,
+                    location=self.location,
                 )
                 state = transfer_run.state
                 self.log.info("Current state is %s", state)

